File: Irrgarten_Java/src/irrgarten/GameState.java
package irrgarten;

/**
 * Clase que almacena informacion sobre el estado del juego
 * 
 * @author Arturo Olivares Martos
 * @author Joaquín Avilés de la Fuente
 */
public class GameState {

    /**
     * Estado del laberinto
     */
    private final String labyrinth;
    /**
     * Estado de los jugadores
     */
    private final String players;
    /**
     * Estado de los monstruos
     */
    private final String monsters;
    /**
     * Identificador del jugador actual
     */
    private final int currentPlayer;
    /**
     * Indicador de si hay un ganador
     */
    private final boolean winner;
    /**
     * Eventos interesantes a lo largo de la partida
     */
    private final String log;

    /**
     * Constructor de la clase GameState
     * @param labyrinth Estado del laberinto
     * @param players Estado de los jugadores
     * @param monsters Estado de los monstruos
     * @param currentPlayer Identificador del jugador actual
     * @param winner Indicador de si hay un ganador
     * @param log Eventos interesantes a lo largo de la partida
     */
    public GameState(String labyrinth, String players, String monsters, 
            int currentPlayer, boolean winner, String log) {
        this.labyrinth = labyrinth;
        this.players = players;
        this.monsters = monsters;
        this.currentPlayer = currentPlayer;
        this.winner = winner;
        this.log = log;
    }

    /**
     * Consultor del estado del laberinto
     * @return Estado del laberinto
     */
    public String getLabyrinth() {
        return labyrinth;
    }
    
    /**
     * Consultor del estado de los jugadores
     * @return Estado de los jugadores
     */
    public String getPlayers() {
        return players;
    }

    /**
     * Consultor del estado de los monstruos
     * @return Estado de los monstruos
     */
    public String getMonsters() {
        return monsters;
    }

    /**
     * Consultor para identificar el jugador actual
     * @return Índice del jugador actual
     */
    public int getCurrentPlayer() {
        return currentPlayer;
    }

    /**
     * Consultor de si ya hay un ganador
     * @return True si hay ganador, false en caso constrario
     */
    public boolean getWinner() {
        return winner;
    }

    /**
     * Consultor de eventos interesantes de la partida
     * @return Eventos interesantes de la partida
     */
    public String getLog() {
        return log;
    }
}

File: Irrgarten_Ruby/game_state.rb
# encoding: UTF-8
module Irrgarten

    # Esta clase permitirá, de forma muy sencilla, almacenar una representación del estado completo del
    # juego: el estado del laberinto, el estado de los jugadores, el estado de los monstruos, el índice del
    # jugador que tiene el turno, un indicador sobre si ya hay un ganador y un atributo adicional para
    # guardar en una cadena de caracteres eventos interesantes que hayan ocurrido desde el turno anterior.
    #
    # @author Joaquin Avilés de la Fuente
    # @author Arturo Olivares Martos
    class GameState

        # Método que devuelve el estado del juego
        #
        # @param labyrinth [String] estado del laberinto
        # @param players [String] estado de los jugadores
        # @param monsters [String] estado de los monstruos
        # @param current_player [int] índice del jugador que tiene el turno
        # @param winner [boolean] Indica si hay ya un ganador o no
        # @param log [String] cadena de caracteres con eventos interesantes
        #
        # @return devuelve el estado del juego
        def initialize (labyrinth, players, monsters, current_player, winner, log)
            @labyrinth=labyrinth
            @players=players
            @monsters=monsters
            @current_player=current_player # Representa el índice del jugador actual
            @winner=winner
            @log=log
        end

        # Método que devuelve el estado del laberinto
        # @return [String] Devuelve el estado del laberinto
        attr_reader :labyrinth

        # Método que devuelve el estado de los jugadores
        # @return [String] Devuelve el estado de los jugadores
        attr_reader :players

        # Método que devuelve el estado de los monstruos
        # @return [String] Devuelve el estado de los monstruos
        attr_reader :monsters

        # Método que devuelve el índice del jugador que tiene el turno
        # @return [int] Devuelve el índice del jugador que tiene el turno
        attr_reader :current_player

        # Método que devuelve si hay un ganador o no
        # @return [boolean] Devuelve si hay un ganador o no
        attr_reader :winner

        # Método que devuelve la cadena de caracteres con eventos interesantes
        # @return [String] Devuelve la cadena de caracteres con eventos interesantes
        attr_reader :log
    end

end
